textattack/datasets/custom_dataset.py

```diff
@@ -0,0 +1,166 @@
+import collections
+import random
+
+import datasets
+import pandas as pd
+
+import textattack
+from textattack.datasets import TextAttackDataset
+
+
+def _cb(s):
+    """Colors some text blue for printing to the terminal."""
+    if not isinstance(s, str):
+        s = "custom " + str(type(s))
+    return textattack.shared.utils.color_text(str(s), color="blue", method="ansi")
+
+
+class CustomDataset(TextAttackDataset):
+    """Loads a Custom Dataset from a file/list of files and prepares it as a
+    TextAttack dataset.
+
+    - name(Union[str, dict, pd.DataFrame]): the user specified dataset file names, dicts or pandas dataframe
```

Comment:
Don't you think `name` argument would be too confusing? Maybe `file_name_or_data`?